client/cli/src/runner.rs

```diff
@@ -27,17 +27,21 @@ use log::info;
 use sc_service::{Configuration, TaskType, TaskManager};
 use sp_utils::metrics::{TOKIO_THREADS_ALIVE, TOKIO_THREADS_TOTAL};
 use std::marker::PhantomData;
+use std::error;
+use std::result;
```

Comment:
nit: I feel that `std::error::Error` in code is more readable in code, even though it's more verbose.
Also, our policy in polkadot was not to shadow `std::result::Result`, so it's always clear if `Result` is `std::result::Result`, but it looks like substrate follow a different strategy.
So I'd be in favor of removing this two use statements and using fully qualified paths as it was previously.

P.S. maybe we won't need `use crate::Result` after this change?

Reply:
The last thing I read was about having one module level plus type being best practice for rust generally speaking.

I am a bit wary of things becoming hard to read, especially with the trait bound `where` statements becoming painfully long.

With the different Result types, it can be repetitive, but on the other hand side, it disambiguates nicely since there is exactly one kind of default/normal `Result<T>` and everything else has `result::Result<T, E>` which improves readability imho.

But I am open for adjustments.

> P.S. maybe we won't need use crate::Result after this change?

I am a bit wary of this, that might cause some ripples for people using it.

Reply:
> With the different Result types, it can be repetitive, but on the other hand side, it disambiguates nicely since there is exactly one kind of default/normal Result<T> and everything else has result::Result<T, E> which improves readability imho.

That's also what I was aiming for. `Result` (aka `std::result::Result`) and `crate::Result`. But the problem is people have different expectations for `Result`.

Reply:
I'll leave this as is for now, since it's merely a matter of taste rather than clear benefit for one over the other. I appreciate the feedback!